[15.0][MIG] template_content_swapper: backport from 16.0 to 15.0

AungKoKoLin1997 opened the pull request:
Backport from 16.0 to 15.0

@qrtl QT4565

AungKoKoLin1997 committed:
[ADD] template_content_swapper

Commit:
[UPD] Update template_content_swapper.pot

OCA-git-bot committed:
[BOT] post-merge updates

AungKoKoLin1997 committed:
[FIX] template_content_swapper
This commit resolves the issue by adding 'report_id', which enables the computation of 'template_id'.
Previously, the problem occurred because 'template_id' is a mandatory field that needs to be calculated before inserting the record.

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (23 of 23 strings)

Translation: server-ux-16.0/server-ux-16.0-template_content_swapper
Translate-URL: https://translation.odoo-community.org/projects/server-ux-16-0/server-ux-16-0-template_content_swapper/it/

AungKoKoLin1997 committed:
[IMP] template_content_swapper: black, isort, prettier

AungKoKoLin1997 committed:
[MIG] template_content_swapper: backport from 16.0 to 15.0

yostashiro reviewed (approved):
Code review and functional test.

kanda999 reviewed (approved):
Code and Functional review.

AungKoKoLin1997 committed:
[FIX] template_content_swapper

This commit fixes the issue where there are no values when the view is rendered by calling _render_template.
For example, when creating a mega menu in the website, the values are None.

kanda999 commented on template_content_swapper/models/ir_ui_view.py:
```diff
@@ -0,0 +1,45 @@
+# Copyright 2024 Quartile Limited
```
```suggestion
# Copyright 2024 Quartile (https://www.quartile.co)
```

kanda999 reviewed (approved):
Functional review.

AungKoKoLin1997 commented:
@oca/server-ux-maintainers 
Can we get this PR merged?

AungKoKoLin1997 committed:
[IMP] template_content_swapper: use _render()

This PR moves the content mapping logic from _render_template to _render, which is used more consistently,
ensuring content replacement works in all rendering scenarios.

Co-authored-by: Tatsuki Kanda <kanda@quartile.co>

yostashiro reviewed (approved):
LGTM.

dreispt commented:
/ocabot merge nobump
